external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaOffsetReader.scala

```diff
@@ -419,6 +416,21 @@ private[kafka010] class KafkaOffsetReader(
     stopConsumer()
     _consumer = null  // will automatically get reinitialized again
   }
+
+  private def getPartitions(): ju.Set[TopicPartition] = {
+    consumer.poll(jt.Duration.ZERO)
+    var partitions = consumer.assignment()
+    val startTimeMs = System.currentTimeMillis()
```

Comment:
For this kind of logic it's better to use `System.nanoTime()` which is monotonic. Also you can do a little less computation this way:

```
val deadline = System.nanoTime() + someTimeout;
while (... && System.nanoTime() < deadline) {

}
```

Reply:
Good point, since @zsxwing suggested new API usage I would wait here and check the Kafka side.